iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/Properties.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (c) 2020, All partners of the iTesla project (http://www.itesla-project.eu/consortium)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.google.common.collect.Sets;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.DanglingLine;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Identifiable.PropertyType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+final class Properties {
```

Comment:
I don't agree with what you have done: it is really not efficient (and the way the maps are run through can create big performance issues...). I think you should do something like this:
```java
Map<String, Property> properties = new HashMap<>();
class Property {
    enum Type { STRING, BOOLEAN, INTEGER, DOUBLE };
    private final Type type;
    private final Object value;
    public Property(int value) { this.type = INTEGER; this.value = value; }
    public Property(String value) { this.type = STRING; this.value = value; }
    public int intValue() {
        if (type == INTEGER) {
            return (int) value;
        }
        throw ...;
    }
    public String stringValue() {
        if (type == STRING) {
             return (String) value;
        }
    }
...
}
```